src/main/java/com/epam/aidial/core/controller/ResourceOperationController.java

```diff
@@ -84,9 +100,75 @@ public Future<?> move() {
         return Future.succeededFuture();
     }
 
+    public Future<?> subscribe() {
+        HttpServerResponse response = context.getResponse();
+        Consumer<ResourceEvent> subscriber = event -> {
```

Comment:
consider moving to the separate method

Reply:
I will leave it as is, it is a pretty small piece of code.

Reply:
After considering I moved it into a method and added permission check since they are liable to change.